tdrs-backend/tdpservice/reports/views.py

```diff
@@ -2,14 +2,36 @@
 import logging
 
 from rest_framework import mixins, viewsets
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from ..users.permissions import CanDownloadReport, CanUploadReport
 
-from ..users.permissions import CanUploadReport
-from .models import User
+from .models import User, ReportFile
 from .serializers import ReportFileSerializer
 
 logger = logging.getLogger()
 logger.setLevel(logging.INFO)
 
+class GetYearList(APIView):
+    """Get list of years for which there are reports."""
+
+    query_string = False
+    pattern_name = "report-list"
+    permission_classes = [CanDownloadReport]
+
+    def get(self, request, **kargs):
+        """Handle get action for get list of years there are reports."""
+        user = request.user
+        is_ofa_admin = user.groups.filter(name="OFA Admin").exists()
+
+        stt_id = kargs.get('stt') if (is_ofa_admin) else user.stt.id
```

Comment:
The way this is written keeps the app from inspecting the user.stt.id field causing an attribute error if an admin doesn't have an stt. The way it was initially written, I added it as the default value to a hash map get operation, so it tried to access that property even if there was a value at the given key in kargs.